src/nvim/normal.c

```diff
@@ -7955,6 +7955,7 @@ static void nv_event(cmdarg_T *cap)
   // later(`f_confirm` or `get_keystroke` for example), but in these cases it is
   // not safe to perform garbage collection because there could be unreferenced
   // lists or dicts being used.
+  finish_op = false;
   may_garbage_collect = false;
   multiqueue_process_events(main_loop.events);
   cap->retval |= CA_COMMAND_BUSY;       // don't call edit() now
```

Comment:
`finish_op = false` should probably go _after_ `multiqueue_process_events(main_loop.events)` . Or maybe before and after ...

Reply:
Sorry, I don't follow ... do you mean in case one of the events on the event queue end up setting `finish_op = true`?

or maybe to let events have access to state telling them that an operator is pending?

Reply:
Moving it after the `multiqueue_process_events(main_loop.events)` allows `nvim_get_mode` to return the correct mode name. Because `getmode()` uses `finish_op=true` to decide whether mode is "o".